witchcraft/server_metrics.go

```diff
@@ -101,7 +101,32 @@ func (s *Server) initMetrics(ctx context.Context, installCfg config.Install) (rR
 		// most up-to-date metric logger is used (s.metricLogger may be updated during initialization).
 		// s.metricLogger is not guaranteed to be non-nil at this point.
 		if s.metricLogger != nil {
-			s.metricLogger.Metric(metricID, metricType, metric1log.Values(valuesToUse), metric1log.Tags(tags.ToMap()))
+			metricTagsParam := metric1log.Tags(tags.ToMap())
+
+			if metricType == "meter" || metricType == "timer" || metricType == "histogram" {
+				mapKey := fmt.Sprintf("%s:%s:%v", metricID, metricType, tags)
```

Comment:
Is it possible that 2 metrics could have equivalent `metricID`, `metricType`, and `tags`, but the tags be in different orders?